env0/resource_environment.go

```diff
@@ -1016,7 +1041,7 @@ func getCreatePayload(d *schema.ResourceData, apiClient client.ApiClientInterfac
 		payload.AutoDeployOnPathChangesOnly = boolPtr(val.(bool))
 	}
 
-	_, isWorkflow := d.GetOk("sub_environment_configuration")
+	isWorkflow := templateType == client.WORKFLOW
```

Comment:
and finally we decide if it's a workflow based on the templateType.
if isWorkflow is True, defaults will be set as needed. This code already exist.
The change is - we decide if it's a workflow based on the template type instead of checking if sub_environment_configuration exist.